### DISCUSSION TITLE
Dynamic instance of Result

### DISCUSSION THREAD

**Discussion by ljimenezalvarez** [vkhorikov/CSharpFunctionalExtensions#388] [Ideas]
Is there a way to create a dynamic instance of Result.Failure(typeof(T), errorMessage) besides of Result.Failure<T>(errorMessage)

**Comment by vkhorikov** [comment]
The library should support dynamics. If you post the code that doesn't work for you, I can take a look.

**Reply by ljimenezalvarez** [reply]
Thank you Vladimir for responding quickly. The thing, Im elaborating a behavior pipeline implemented in the [Mediatr ](https://github.com/jbogard/MediatR) library.

I'm not sure, something like this:

```
>  public async Task<TResponse> Handle(TRequest request, CancellationToken cancellationToken,
>             RequestHandlerDelegate<TResponse> next)
>         {
>             try
>             {
>                 return await next();
>             }
>             catch (Exception ex)
>             {
>                 var msg = ex.GetBaseException().Message;
>                 _logger.LogError(msg);
> 
>                 if (typeof(TResponse) != typeof(Result) &&
>                     typeof(TResponse).GetGenericTypeDefinition() != typeof(Result<>))
>                 {
>                     return Result.Failure(typeof(TGenericTypeDefinition), msg) as TResponse;
>                 }
> 
>                 throw;
>             }
>         }
>     }
```

Thank you for your hard work, contribution and efforts for the software industry.

Regards

**Comment by vkhorikov** [comment]
Sorry for the delay. Could you please boil it down to a failing test?

Also, check out this topic, it might be helpful for your use case: https://github.com/vkhorikov/CSharpFunctionalExtensions/issues/322

**Comment by ljimenezalvarez** [comment]
Thank you Vladimir
